pkg/limitador/k8s_objects.go

```diff
@@ -66,6 +67,18 @@ func Deployment(limitador *limitadorv1alpha1.Limitador, deploymentOptions Deploy
 		image = *limitador.Spec.Image
 	}
 
+	// setting the desired labels, the hardcoded ones and the limitador ones.
+	mutableLabels := Labels(limitador)
+	if limitador.ObjectMeta.Labels != nil {
```

Comment:
no need to check `imitador.ObjectMeta.Labels != nil`. It is safe to iterate over map which is a reference to nil.